storage/entities/base_entity.js

```diff
@@ -305,33 +305,44 @@ class BaseEntity {
 	parseFilters(filters) {
 		let filterString = null;
 
-		const parseFilterObject = object =>
-			Object.keys(object)
-				.map(key => this.filters[key])
-				.join(' AND ');
-
 		if (Array.isArray(filters)) {
-			filterString = filters
-				.map(filterObject => parseFilterObject(filterObject))
-				.join(' OR ');
-		} else if (typeof filters === 'object') {
-			filterString = parseFilterObject(filters);
+			filterString = this._parseArrayFilter(filters);
+		} else if (Object.keys(filters).length > 0) {
+			filterString = this._parseObjectFilter(filters);
 		}
 
-		let filtersObject = filters;
+		const filtersObject = this._filtersToObject(filters);
 
-		if (Array.isArray(filters)) {
-			filtersObject = Object.assign({}, ...filters);
-		}
+		return this._filterClause(filterString, filtersObject);
+	}
 
+	_filterClause(filterString, filtersObject) {
+		let filterClause;
 		if (filterString) {
-			return `WHERE ${this.adapter.parseQueryComponent(
+			filterClause = `WHERE ${this.adapter.parseQueryComponent(
 				filterString,
 				filtersObject
 			)}`;
 		}
 
-		return '';
+		return filterClause || '';
+	}
+
+	// eslint-disable-next-line class-methods-use-this,no-unused-vars
```

Comment:
Why `no-unused-vars`? Shouldn't it be `class-methods-use-this` only?

Reply:
Done